stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/FileAttachmentsView.kt

```diff
@@ -230,31 +226,18 @@ private class FileAttachmentViewHolder(
 
     private fun handleInProgressAttachment(fileSizeView: TextView) {
         attachment?.let { attachment ->
-            attachment.uploadId?.let(ProgressTrackerFactory::getOrCreate)?.let { tracker ->
-                val progress = tracker.currentProgress()
-                val completion = tracker.isComplete()
-                val totalValue = MediaStringUtil.convertFileSizeByteCount(attachment.upload?.length() ?: 0)
-
-                val fileProgress = progress.combine(completion, ::Pair)
-
-                clearScope()
-                scope = CoroutineScope(DispatcherProvider.Main)
-
-                scope?.launch {
-                    fileProgress.collect { (progress, isComplete) ->
-                        val uploadedBytes = (progress / 100F * tracker.maxValue).toLong()
-                        updateProgress(
-                            context,
-                            fileSizeView,
-                            binding.progressBar,
-                            attachment,
-                            uploadedBytes,
-                            isComplete,
-                            totalValue
-                        )
-                    }
-                }
-            }
+            val uploadState = (attachment.uploadState as Attachment.UploadState.InProgress)
+            val progress = uploadState.currentProgress
+            val fileLength = attachment.upload?.length() ?: 0
+            val totalValue = MediaStringUtil.convertFileSizeByteCount(fileLength)
+            val nominalProgress = progress * fileLength / 100f
+
+            fileSizeView.text =
```

Comment:
Not specific to this line, but the entire message item slides up and down a bit vertically as all of these progress updates are happening, which wasn't the case with the previous implementation.

Reply:
Yes, It's a known UI glitch. Working on it.

Reply:
I'm still seeing this problem with the latest version of this PR

Reply:
https://user-images.githubusercontent.com/12054216/135230527-fe349c8c-033a-46e0-99a8-d7ff36d14013.mp4


https://user-images.githubusercontent.com/12054216/135230539-35463d29-cfc2-4ae3-889f-441c1039d22e.mp4

Reply:
One of the issues is that the ViewHolder resizes vertically really quickly, dropping down below the Message Input briefly. Then it tries to animate / climb back up to its normal position, but I think the rapid updates happening to the list keep interrupting that animation, hence the laggy movement upwards.